presto-main/src/main/java/com/facebook/presto/split/RemoteSplit.java

```diff
@@ -31,18 +30,18 @@
 public class RemoteSplit
         implements ConnectorSplit
 {
-    private final URI location;
+    private final String location;
```

Comment:
Do we wanna create an abstraction called `Location` rather than using raw string? I'm thinking to replace the `URI` interfaces in `LocationFactory` all with `Location`.